kernel/kernel.cpp

```diff
@@ -4,6 +4,20 @@
 #include "mmio.hpp"
 #include "uart.hpp"
 
+extern "C" // Needed by clang
```

Comment:
Not only by Clang, I suppose.

Nevermind, maybe we should move it to another file. But for now, it should be enough.